Source/WebKit/UIProcess/Automation/SimulatedInputDispatcher.cpp

```diff
@@ -453,13 +453,14 @@ void SimulatedInputDispatcher::transitionInputSourceToState(SimulatedInputSource
 
             b.location = location;
 
-            if (!a.scrollDelta->isZero())
-                b.scrollDelta->contract(a.scrollDelta->width(), a.scrollDelta->height());
+            auto usedScrollDelta = b.scrollDelta.value_or(WebCore::IntSize());
+            if (a.scrollDelta && !a.scrollDelta->isZero())
+                usedScrollDelta.contract(a.scrollDelta->width(), a.scrollDelta->height());
 
-            if (!b.scrollDelta->isZero()) {
+            if (!usedScrollDelta.isZero()) {
                 LOG(Automation, "SimulatedInputDispatcher[%p]: simulating Wheel from (%d, %d) to (%d, %d) for transition to %d.%d", this, a.scrollDelta->width(), a.scrollDelta->height(), b.scrollDelta->width(), b.scrollDelta->height(), m_keyframeIndex, m_inputSourceStateIndex);
```

Comment:
Does this logging also need to be taken care of?
```suggestion
                LOG(Automation, "SimulatedInputDispatcher[%p]: simulating Wheel from (%d, %d) to (%d, %d) for transition to %d.%d", this, a.scrollDelta->width(), a.scrollDelta->height(), usedScrollDelta.width(), usedScrollDelta.height(), m_keyframeIndex, m_inputSourceStateIndex);
```

Reply:
we also probably cant assume that `a.scrollDelta` exists either maybe we should just default both values
```cpp
auto aScrollDelta = a.scrollDelta.value_or(IntSize());
auto bScrollDelta = b.scrollDelta.value_or(IntSize());
```